From 82c6d30085ef38fcd97abc5e5293f9f4644f3092 Mon Sep 17 00:00:00 2001
From: Seth Falcon <seth@userprimary.net>
Date: Thu, 1 Jul 2010 20:09:57 -0700
Subject: [PATCH 1/1] Fix ps command in riak stop for OS X

OS X's ps command does not support -f, might as well
pull out just the fields needed using -o
---
 rel/files/riak |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rel/files/riak b/rel/files/riak
index d425550..a4291fa 100755
--- a/rel/files/riak
+++ b/rel/files/riak
@@ -70,7 +70,7 @@ case "$1" in
 
     stop)
         # Wait for the node to completely stop...
-        PID=`ps -ef|grep "$RUNNER_BASE_DIR/.*/[b]eam.smp|awk '{print $2}'"`
+        PID=`ps -Ao pid,comm|grep "$RUNNER_BASE_DIR/.*/[b]eam.smp|awk '{print $1}'"`
         $NODETOOL stop
         while `kill -0 $PID 2>/dev/null`;
         do
-- 
1.7.1

